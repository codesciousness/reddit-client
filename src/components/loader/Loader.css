.Loader {
    display: flex;
    justify-content: center;
    align-items: center;
}

.Loader__1 {
    display: flex;
    justify-content: center;
    align-items: center;
    border: 1px solid transparent;
    border-radius: 50%;
    border-top-color: rgb(50, 65, 85);
    border-bottom-color: rgb(50, 65, 85);
    width: 60px;
    height: 60px;
    -webkit-animation: rotate 1.5s linear infinite;
    animation: rotate 1.5s linear infinite;
}

.Loader__2 {
    display: flex;
    justify-content: center;
    align-items: center;
    border: 1px solid transparent;
    border-radius: 50%;
    border-top-color: lightsteelblue;
    border-bottom-color: lightsteelblue;
    width: 50px;
    height: 50px;
    -webkit-animation: rotate-reverse 1s linear infinite;
    animation: rotate-reverse 1s linear infinite;
}

.Loader__3 {
    border: 1px solid transparent;
    border-radius: 50%;
    border-top-color:rgb(50, 65, 85);
    border-bottom-color: rgb(50, 65, 85);
    width: 40px;
    height: 40px;
    -webkit-animation: rotate 600ms linear infinite;
    animation: rotate 600ms linear infinite;
}

@-webkit-keyframes rotate {
    100% { -webkit-transform: rotate(360deg); }
}

@keyframes rotate {
    100% { transform: rotate(360deg); }
}

@-webkit-keyframes rotate-reverse {
    100% { -webkit-transform: rotate(-360deg); }
}

@keyframes rotate-reverse {
    100% { transform: rotate(-360deg); }
}