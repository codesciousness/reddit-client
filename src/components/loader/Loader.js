import React from 'react';
import './Loader.css';

const Loader = () => {
    return (
        <section className="Loader">
            <div className="Loader__1">
                <div className="Loader__2">
                    <div className="Loader__3">
                    </div>
                </div>
            </div>
        </section>
    );
}

export default Loader;