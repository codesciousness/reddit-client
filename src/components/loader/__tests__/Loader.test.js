import React from 'react';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import Loader from '../Loader';

describe('<Loader />', () => {
    test('Loader component is defined', () => {
        expect(<Loader />).toBeDefined();
    });
    test('matches the snapshot', () => {
        const component = create(<Loader />);
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('renders correctly', () => {
        const { container } = render(<Loader />);
        expect(container.querySelector('.Loader')).toBeInTheDocument();
    });
});