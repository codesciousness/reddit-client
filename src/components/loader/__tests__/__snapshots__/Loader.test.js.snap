// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> matches the snapshot 1`] = `
<section
  className="Loader"
>
  <div
    className="Loader__1"
  >
    <div
      className="Loader__2"
    >
      <div
        className="Loader__3"
      />
    </div>
  </div>
</section>
`;
